src/test/scala/firrtlTests/InfoSpec.scala

```diff
@@ -172,4 +172,27 @@ class InfoSpec extends FirrtlFlatSpec with FirrtlMatchers {
     val expectedInfos = Seq(FileInfo(StringLit("Top.scala 15:14")), FileInfo(StringLit("myfile.fir 6:4")))
     circuitState should containTree { case MultiInfo(`expectedInfos`) => true }
   }
+
+  "FileInfo" should "be able to contain a escaped characters" in {
+    def input(anno: String): String =
+      s"""circuit m: @[$anno]
+        |  module m:
+        |    skip
+        |""".stripMargin
+    def parseAnno(anno: String): FileInfo = {
+      firrtl.Parser.parse(input(anno)).info.asInstanceOf[FileInfo]
+    }
+
+    parseAnno("test\\\ntest").escaped should be ("test\\\ntest")
+    parseAnno("test\\\ntest").unescaped should be ("test\ntest")
+    parseAnno("test\\\ttest").escaped should be ("test\\\ttest")
+    parseAnno("test\\\ttest").unescaped should be ("test\ttest")
+    parseAnno("test\\\\test").escaped should be ("test\\\\test")
+    parseAnno("test\\\\test").unescaped should be ("test\\test")
+    parseAnno("test\\]test").escaped should be ("test\\]test")
+    parseAnno("test\\]test").unescaped should be ("test]test")
+    parseAnno("test[\\][\\]test").escaped should be ("test[\\][\\]test")
+    parseAnno("test[\\][\\]test").unescaped should be ("test[][]test")
+  }
```

Comment:
1. `parseAnno` is the wrong name here (I'm guessing it's from copy-paste), but just like `parse` would be better or `parseInfo`.

2. I would like to see tests that use the actual ANTLR Parser (because this changes the emitted FIRRTL) as well as in and out of protobuf.

Reply:
1. I am going to change it to `parseInfo`. I didn't consider that `anno` was already used for something else.

2. Isn't `firrtl.Parser.parse` invoking the antlr parser? I can add some protobuf tests.